### PR TITLE
Add note about no warnings being shown for shorthand form xref links

### LINKED ISSUE #1142: Show warning if xref is not found
### Functional impact
Using an invalid xref link like `@does-not-exist` in a contextual file does not trigger a warning on build like standard markdown links do. 

### Minimal repro steps
1. Add an invalid xref link to a document, eg. `@does-not-exist`
2. Run `docfx`

### Expected result
I would expect a warning that no xref for `@does-not-exist` was found.

### Actual result
Only a info is shown how many external references are found. Something like:

```
Info: [Build Document.LinkPhaseHandler.UpdateContext]0 external references found in 3 xref maps.
```

### PR DESCRIPTION
**Pull request by pascalberger**
Fixes #1142

### CODE REVIEW

File: Documentation/tutorial/links_and_cross_references.md [hunk 1]
```diff
@@ -188,6 +188,10 @@ You can also use `@uid_to_another_file` to quickly reference another file. There
 
 The render result of `@` form is same as auto link form. For example, `@System.String` is same as `<xref:System.String>`.
 
+> [!Warning]
+> Since `@` is a common character in a document, DocFx won't show a warning if UID wasn't found for a shorthand form xref link.
```
Comment (superyyrrzz) [inline Documentation/tutorial/links_and_cross_references.md]: DocFx -> DocFX

File: Documentation/tutorial/links_and_cross_references.md [hunk 2]
```diff
@@ -188,6 +188,10 @@ You can also use `@uid_to_another_file` to quickly reference another file. There
...
+> Warnings for missing links are shown though for auto links and Markdown links.
```
Comment (vicancy) [inline Documentation/tutorial/links_and_cross_references.md]: ~~though~~

File: Documentation/tutorial/links_and_cross_references.md (hunk 1 above)
Comment (vicancy) [inline Documentation/tutorial/links_and_cross_references.md]: won't => doesn't, wasn't => isn't ?

Comment (pascalberger) [inline Documentation/tutorial/links_and_cross_references.md]: Fixed

File: Documentation/tutorial/links_and_cross_references.md (hunk 2 above)
Comment (pascalberger) [inline Documentation/tutorial/links_and_cross_references.md]: Fixed

**Pull request merged**